File: Remove Linked List Elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    /**
     * @param head a ListNode
     * @param val an integer
     * @return a ListNode
     */
    ListNode *removeElements(ListNode *head, int val) {
        // Write your code here
        ListNode dummy(INT_MIN);
        dummy.next = head;
        ListNode *pre = &dummy;
        ListNode *post = head;
        while (post)
        {
            if (post->val == val)
            {
                auto tmp = post;
                post = post->next;
                pre->next = post;
                delete tmp;
            }
            else
            {
                post = post->next;
                pre = pre->next;
            }
        }
        return dummy.next;
        
    }
};

File: Remove Linked List Elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param head, a ListNode
    # @param val, an integer
    # @return a ListNode
    def removeElements(self, head, val):
        # Write your code here
        dummy = ListNode(-1)
        cur = dummy
        while head:
            if head.val != val:
                cur.next, cur = head, head
            head = head.next
        cur.next = None
        return dummy.next